superset-frontend/src/visualizations/TimeTable/TimeTable.jsx

```diff
@@ -209,10 +209,20 @@ const TimeTable = ({
             .reduce((a, b) => a + b);
       } else if (column.colType === 'avg') {
         // Average over the last {timeLag}
-        v =
-          reversedEntries
-            .map((k, i) => (i < column.timeLag ? k[valueField] : 0))
-            .reduce((a, b) => a + b) / column.timeLag;
+        v = null;
+        if (reversedEntries.length > 0) {
+          const stats = reversedEntries.slice(undefined, column.timeLag).reduce(
+            function ({ count, sum }, entry) {
+              return entry[valueField] != null
```

Comment:
Just to be sure, let's also check for `undefined` here

Reply:
Thank you for rating.
This piece is specially written to check only nulls to maintain speed.  I thought that there was no need for extra checks on a large tables.  In practice, it seems to work . 

Do we really need to do this?

Reply:
@pavlozt if this extra check has a significant performance impact then I'm ok leaving it out. But I would be surprised if that's the case. The reason I was hoping to add broader checking is because many of these legacy plugins lack proper type declarations, hence it makes sense to do extra checking when unsure what the types may be.

Reply:
Ok. Additional check added and and re-tested.